api/types/types.go

```diff
@@ -525,3 +529,30 @@ type Runtime struct {
 	Path string   `json:"path"`
 	Args []string `json:"runtimeArgs,omitempty"`
 }
+
+// DiskUsage contains response of Remote API:
+// GET "/system/df"
+type DiskUsage struct {
+	LayersSize int64
+	Images     []*Image
+	Containers []*Container
+	Volumes    []*Volume
+}
+
+// PruneRequest contains the options for Remote API:
+// POST "/system/prune"
+type PruneRequest struct {
+	PruneContainers    bool
+	PruneVolumes       bool
+	PruneImages        bool
+	DanglingImagesOnly bool
+}
```

Comment:
Instead of having just a single endpoint for prune, I think it might be better to have one endpoint per type (matching the cli).  

The request data is going to expand a lot with different filtering options, and I think it's easier to represent that in a sane way by having a request type for each prune-able object.